### ISSUE TITLE
Rebuilding description page for illustrations of a book

### ISSUE THREAD

**Issue by alexbrollo** [danmichaelo/croptool#115]
Dealing with multipage files (pdf, djvu) wikisource projects can effectively use CropTool to extract illustrations, with excellent results and lots of time saving.

Nevertheless, resulting files are something completely different from original files, the latter being "books" and the former being "illustrations fron a book". Usually metadata for books are stores into a Template:Book while metadata for illustrations should be stored into a Template:Info; author is often different, since the author of illustrations is stored into a parameter Illustrator of Template:Book; categorization is completely different.

I'll try to build a javascript algorithm to convert metadata of a book into metadata for its illustrations; nevertheless I think that resulting text need some editing before upload, so my suggestion is to add a textbox to CropTool, to allow review/editing of the whole wikitext of description page.

**Comment by danmichaelo** [comment]
I'm open to that idea. It would probably be a good idea if all users were forced to review the metadata before submitting, even if it meant just looking at it and pressing "ok" :)

Need to find some time to implement it though. Btw. do you often run into the problem of CropTool not responding? It seems to happen more often recently (https://phabricator.wikimedia.org/T198503) and I struggle finding out how to fix it..

**Comment by alexbrollo** [comment]
CropTool often didn't respond some weeks ago, by now it runs perfectly for me. 

Another suggestion: when you'll add an edit textbox for wikicode of description page, please save edited version of wikicode so that it can be used again and again for illustrations of the same book if possible.

**Comment by danmichaelo** [comment]
> Another suggestion: when you'll add an edit textbox for wikicode of description page, please save edited version of wikicode so that it can be used again and again for illustrations of the same book if possible. 

Hm, I'm a bit afraid that could cause users to save a completely wrong wikitext when working with different images. Should still be quite efficient to have the text in a separate editor and copy-paste it fom there, shouldn't it?

**Comment by alexbrollo** [comment]
Yes, you are right, I came to the same conclusion just some minutes after posting my suggestion :-)
A simple textbox where to copy an external wikitext will be largely sufficient.
